src/api/app/models/group.rb

```diff
@@ -201,6 +204,10 @@ def involved_projects_ids
     ### all projects where user is maintainer
     Relationship.projects.where(group_id: id, role_id: role.id).distinct.pluck(:project_id)
   end
+
+  def send_metric_for_group
+    RabbitmqBus.send_to_bus('metrics', "group count=#{Group.count}")
```

Comment:
- Potentially this event will happen on year apart
- Potentially this event will not happen for a year

How can I know how many groups are there yesterday?

Reply:
Yep. With this approach, after introducing this change in the code, there will be no data until a group is created or destroyed. The advantage, however, is that there is no need to query periodically for data that has not been changed. To prevent this from happening, could we introduce kind of a "data migration" to store the first measurement, after this code is deployed? :thinking:

> How can I know how many groups are there yesterday?

Once the first measurement is stored, `fill(previous)` from Grafana should provide the value required.

Reply:
> The advantage, however, is that there is no need to query periodically for data that has not been changed. 

How is this an advantage?

> Once the first measurement is stored, fill(previous) from Grafana should provide the value required.

Only if my time range includes the previous point.

Reply:
@hennevogel it wouldn't be so useful to every hour, have a cronjob pushing `Group.count` to influxdb. This for sure would fit "measures that don't make sense and we should drop to save space". 

Every measure should tell us a use case or be used to tell a story. The scenario that we saw as necessary was: "The number of groups are changing in an unexpected way, let's alert if it grows or shrinks abruptly". 

Similar scenario we imagined for users in beta.  Here, we want to know from time to time (i.e after a new feature is released under beta), if the user are moving from/to beta. That's would help us to understand if user are happy with beta, or moving away from that because of the new feature. To have cronjob pushing aggregated data about Group/Users, blindly sounds, IMO,  unnecessary

Reply:
We still measure each time a group is created or deleted, but we introduced a monthly clockwork entry to ensure the following:
 1. we are going to have a first measurement as soon as this code is deployed for the first time, and
 1. there is at least one measurement per month.

What do you think about this solution?

Reply:
I still don't see what we want to monitor with that. Possible scenario: If an admin is spamming the application with new groups or deleting wrongly a huge lot of groups, as demolition squad,  i want to know about it ASAP and not after a month..